packages/arb-token-bridge-ui/src/components/TransactionHistory/useIndexerHistory.ts

```diff
@@ -0,0 +1,210 @@
+import { useMemo } from 'react'
+import { Address } from 'viem'
+import useSWRImmutable from 'swr/immutable'
+import useSWR from 'swr'
+import { getProviderForChainId } from '@/token-bridge-sdk/utils'
+import { useArbitrumIndexer } from '../../../../indexer-provider'
+import { AssetType } from '../../hooks/arbTokenBridge.types'
+import { DepositStatus, MergedTransaction } from '../../state/app/state'
+import { BigNumber, utils } from 'ethers'
+import { ether } from '../../constants'
+import {
+  sortByTimestampDescending,
+  UseTransactionHistoryResult
+} from '../../hooks/useTransactionHistory'
+import { isExperimentalFeatureEnabled } from '../../util'
+import { useTokensFromLists } from '../TransferPanel/TokenSearchUtils'
+import { fetchErc20Data } from '../../util/TokenUtils'
+
+type PartialTransfer = {
+  fromAddress: string
+  toAddress: string
+  timestamp: bigint
+  executionTimestamp: bigint
+  status: string
+  amount: bigint
+  txHash: string
+  childChainId: number
+  parentChainId: number
+}
+
+type EthIndexerTransfer = PartialTransfer & {
+  type: 'ETH'
+  tokenAddress: undefined
+}
+
+type Erc20IndexerTransfer = PartialTransfer & {
+  type: 'ERC20'
+  tokenAddress: string
+}
+
+type IndexerTransfer = EthIndexerTransfer | Erc20IndexerTransfer
+
+function getIndexerTransferStatus(tx: IndexerTransfer) {
+  switch (tx.status) {
+    case 'PARENT_CHAIN_CONFIRMED':
+    case 'CHILD_CHAIN_REDEMPTION_SCHEDULED':
+      return 'pending'
+    case 'CHILD_CHAIN_EXECUTED':
+      return 'success'
+    default:
+      return 'failure'
+  }
+}
+
+function getIndexerDepositStatus(tx: IndexerTransfer): DepositStatus {
+  switch (tx.status) {
+    case 'PARENT_CHAIN_CONFIRMED':
+      return DepositStatus.L1_PENDING
+    case 'CHILD_CHAIN_REDEMPTION_SCHEDULED':
+      return DepositStatus.L2_PENDING
+    case 'CHILD_CHAIN_EXECUTED':
+      return DepositStatus.L2_SUCCESS
+    default:
+      return DepositStatus.L2_FAILURE
+  }
+}
+
+type TokenDetails = {
```

Comment:
Indexer only gives us token address because it would take a long time to fetch token data and store it. We need to fetch the token data ourselves. 

see L162